src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -2347,10 +2354,12 @@ private function step_in_table(): bool {
 				) {
 					return $this->step();
 				}
+
+				// This FORM is special because it immediately closes and cannot have other children.
+				$this->state->current_token->has_self_closing_flag = true;
```

Comment:
This is where I'm pondering.

Reply:
What's the motivation for doing it this way instead of adding and popping right here?